src/bll/weblinklistener/httphostrequest.ts

```diff
@@ -0,0 +1,98 @@
+import {HttpRequestData} from "./httprequestdata";
+import {WorkspaceProxy} from "../moduleproxies/workspace-proxy";
+import * as path from "path";
+import {FsProxy} from "../moduleproxies/fs-proxy";
+import {Logger} from "../utils/logger";
+import {inject, injectable} from "inversify";
+import {TYPES} from "../utils/constants";
+import {WindowProxy} from "../moduleproxies/window-proxy";
+import {UriProxy} from "../moduleproxies/uri-proxy";
+import {Utils} from "../utils/utils";
+
+@injectable()
+export class HttpHostRequest {
+
+    public constructor(@inject(TYPES.WorkspaceProxy) private workspaceProxy: WorkspaceProxy,
+                       @inject(TYPES.FsProxy) private fsProxy: FsProxy,
+                       @inject(TYPES.UriProxy) private uriProxy: UriProxy,
+                       @inject(TYPES.WindowProxy) private windowProxy: WindowProxy) {
+    }
+
+    public async processRequest(bytes: Uint8Array): Promise<{succeed: boolean, httpVersion: string}> {
+        const input: string[] = bytes.toString().trim().split("\n");
+        const firstLine: string = input[0];
+        Logger.logDebug(`HttpHostRequest#processRequest: firstLine is ${firstLine}`);
+        const tokens: string[] = firstLine.trim().split(/[\t\r\n\f ]+/);
+        if (tokens.length < 3) {
+            Logger.logWarning(`HttpHostRequest#processRequest: not enough tokens: ${tokens.join(";")}`);
+            return {succeed: false, httpVersion: undefined};
+        }
+
+        const uri = tokens[1];
+        const httpVersion = tokens[2];
+        let path = uri;
+        const params = {};
+
+        const endOfPathIndex: number = uri.indexOf("?");
+        if (endOfPathIndex >= 0) {
+            path = uri.substring(0, endOfPathIndex);
+            const request = uri.substring(endOfPathIndex + 1);
+            this.parseGetRequestParams(request, params);
+        }
+
+        const succeed = await this.fireRequestAccepted(path, params);
+        return {succeed: succeed, httpVersion: httpVersion};
+    }
+
+    private parseGetRequestParams(encodedURIComponent: string, pars: {}): void {
```

Comment:
To parse query parameters could be used `url.parse` method like that: https://stackoverflow.com/a/8590367/305875